### PR TITLE
Update to .NET 9

### PR DESCRIPTION
**Pull request by brminnick**
### Description of Change ###

This PR updates `CommunityToolkit.Maui.Markup` for .NET 9

 ### PR Checklist ###
 
 - [x] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [x] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/480

 ### Breaking Changes ###
- String-based bindings are no-longer trim safe
  - eg `.Bind(Label.TextProperty, nameof(MyViewModel.LabelText)` is not trim safe
  - Use `.Bind(Label.TextProperty, static (MyViewModel vm) => vm.LabelText)` instead
- `.BindClickGesture` Removed
  - Use `.BindTapGesture` instead
- `.ClickGesture` Removed
  - Use `.TapGesture` instead

### Benchmarks ###

**.NET 8**
| Method                                       | Mean     | Error     | StdDev    | Ratio | RatioSD | Allocated | Alloc Ratio |
|--------------------------------------------- |---------:|----------:|----------:|------:|--------:|----------:|------------:|
| ExecuteDefaultBindings_ViewModelToView       | 9.947 ns | 0.1944 ns | 0.1818 ns |  1.00 |    0.02 |         - |          NA |
| ExecuteDefaultBindingsMarkup_ViewModelToView | 9.857 ns | 0.1067 ns | 0.0946 ns |  0.99 |    0.02 |         - |          NA |
| ExecuteTypedBindingsMarkup_ViewModelToView   | 9.965 ns | 0.1869 ns | 0.1748 ns |  1.00 |    0.02 |         - |          NA |

| Method                                       | Mean     | Error    | StdDev  | Ratio | RatioSD | Gen0   | Allocated | Alloc Ratio |
|--------------------------------------------- |---------:|---------:|--------:|------:|--------:|-------:|----------:|------------:|
| ExecuteDefaultBindings_ViewToViewModel       | 707.6 ns |  6.35 ns | 5.94 ns |  1.00 |    0.01 | 0.0076 |     160 B |        1.00 |
| ExecuteDefaultBindingsMarkup_ViewToViewModel | 694.7 ns |  7.92 ns | 7.02 ns |  0.98 |    0.01 | 0.0076 |     160 B |        1.00 |
| ExecuteTypedBindingsMarkup_ViewToViewModel   | 698.3 ns | 10.45 ns | 9.78 ns |  0.99 |    0.02 | 0.0076 |     160 B |        1.00 |

| Method                          | Mean      | Error    | StdDev   | Ratio | RatioSD | Gen0   | Gen1   | Allocated | Alloc Ratio |
|-------------------------------- |----------:|---------:|---------:|------:|--------:|-------:|-------:|----------:|------------:|
| InitializeDefaultBindings       |  10.13 μs | 0.177 μs | 0.204 μs |  1.00 |    0.03 | 0.1373 | 0.1221 |   2.53 KB |        1.00 |
| InitializeDefaultBindingsMarkup |  10.18 μs | 0.196 μs | 0.269 μs |  1.01 |    0.03 | 0.1373 | 0.1221 |   2.53 KB |        1.00 |
| InitializeTypedBindingsMarkup   | 138.97 μs | 2.692 μs | 2.518 μs | 13.72 |    0.36 | 0.4883 | 0.2441 |  11.06 KB |        4.37 |

**.NET 9**

| Method                                       | Mean      | Error    | StdDev   | Ratio | RatioSD | Allocated | Alloc Ratio |
|--------------------------------------------- |----------:|---------:|---------:|------:|--------:|----------:|------------:|
| ExecuteDefaultBindings_ViewModelToView       |  18.52 ns | 0.212 ns | 0.199 ns |  1.00 |    0.01 |         - |          NA |
| ExecuteDefaultBindingsMarkup_ViewToViewModel | 111.60 ns | 0.246 ns | 0.218 ns |  6.03 |    0.06 |         - |          NA |
| ExecuteTypedBindingsMarkup_ViewModelToView   |  18.39 ns | 0.169 ns | 0.158 ns |  0.99 |    0.01 |         - |          NA |

| Method                                       | Mean      | Error    | StdDev   | Ratio | Allocated | Alloc Ratio |
|--------------------------------------------- |----------:|---------:|---------:|------:|----------:|------------:|
| ExecuteDefaultBindings_ViewToViewModel       | 112.15 ns | 0.225 ns | 0.199 ns |  1.00 |         - |          NA |
| ExecuteDefaultBindingsMarkup_ViewModelToView |  18.59 ns | 0.200 ns | 0.187 ns |  0.17 |         - |          NA |
| ExecuteTypedBindingsMarkup_ViewToViewModel   | 115.63 ns | 0.225 ns | 0.211 ns |  1.03 |         - |          NA |

| Method                          | Mean       | Error     | StdDev    | Ratio | RatioSD | Gen0   | Gen1   | Allocated | Alloc Ratio |
|-------------------------------- |-----------:|----------:|----------:|------:|--------:|-------:|-------:|----------:|------------:|
| InitializeDefaultBindings       |   7.798 μs | 0.0847 μs | 0.0793 μs |  1.00 |    0.01 | 0.1068 | 0.0916 |   2.03 KB |        1.00 |
| InitializeDefaultBindingsMarkup |   7.735 μs | 0.1249 μs | 0.1107 μs |  0.99 |    0.02 | 0.1068 | 0.0916 |   2.03 KB |        1.00 |
| InitializeTypedBindingsMarkup   | 126.444 μs | 0.7803 μs | 0.7299 μs | 16.22 |    0.18 | 0.4883 | 0.2441 |  10.64 KB |        5.24 |

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ variables:
   PreviewNumber: $[counter(variables['CurrentSemanticVersionBase'], 1001)]
   CurrentSemanticVersion: '$(CurrentSemanticVersionBase)-preview$(PreviewNumber)'
   NugetPackageVersion: '$(CurrentSemanticVersion)'
-  NET_VERSION: '8.0.x'
+  NET_VERSION: '9.0.100-rc.2.24474.11'
```
Comment (brminnick) [inline azure-pipelines.yml:6]: Change this to 9.0.x before merging

```suggestion
  NET_VERSION: '9.0.x'
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -66,7 +66,7 @@ jobs:
         inputs:
           packageType: 'sdk'
           version: '$(NET_VERSION)'
-          includePreviewVersions: false
+          includePreviewVersions: true
```
Comment (brminnick) [inline azure-pipelines.yml:69]: Change to false before merging
```suggestion
          includePreviewVersions: false
```

File: samples/CommunityToolkit.Maui.Markup.Sample/CommunityToolkit.Maui.Markup.Sample.csproj [hunk 3]
```diff
@@ -52,14 +58,14 @@
...
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Maui.Controls" Version="$(MauiPackageVersion)"/>
-    <PackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="$(MauiPackageVersion)"/>
+    <PackageReference Include="Microsoft.Maui.Controls" Version="$(MauiPackageVersion)" />
+    <PackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="$(MauiPackageVersion)" />
     <PackageReference Include="CommunityToolkit.Maui" Version="$(MauiCommunityToolkitPackageVersion)" />
     <PackageReference Include="CommunityToolkit.Mvvm" Version="8.3.2" />
-    <PackageReference Include="Microsoft.Extensions.Http.Resilience" Version="8.10.0" />
+    <PackageReference Include="Microsoft.Extensions.Http.Resilience" Version="9.0.0-preview.9.24507.7" />
```
Comment (brminnick) [inline samples/CommunityToolkit.Maui.Markup.Sample/CommunityToolkit.Maui.Markup.Sample.csproj:65]: Reminder to update Microsoft.Extensions.Http.Resilience to stable (if available on launch day)

```suggestion
    <PackageReference Include="Microsoft.Extensions.Http.Resilience" Version="9.0.0" />
```

File: Directory.Build.props [hunk 4]
```diff
@@ -1,25 +1,30 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Project>
   <PropertyGroup>
-    <NetVersion>net8.0</NetVersion>
-    <MauiPackageVersion>8.0.93</MauiPackageVersion>
+    <NetVersion>net9.0</NetVersion>
+    <MauiPackageVersion>9.0.0-rc.2.24503.2</MauiPackageVersion>
```
Comment (brminnick) [inline Directory.Build.props:5]: Reminder to update this to the latest stable version of Microsoft.Maui.Controls before merging

File: Directory.Build.props [hunk 5]
```diff
@@ -1,25 +1,30 @@
...
     <MauiCommunityToolkitPackageVersion>9.1.1</MauiCommunityToolkitPackageVersion>
-    <LangVersion>latest</LangVersion>
+    <LangVersion>preview</LangVersion>
```
Comment (brminnick) [inline Directory.Build.props:7]: Reminder to revert back to `latest` before merging

```suggestion
    <LangVersion>latest</LangVersion>
```

File: samples/CommunityToolkit.Maui.Markup.Sample/CommunityToolkit.Maui.Markup.Sample.csproj [hunk 6]
```diff
@@ -22,20 +22,23 @@
...
+    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">15.0</SupportedOSPlatformVersion>
+    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">15.0</SupportedOSPlatformVersion>
     <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">21.0</SupportedOSPlatformVersion>
     <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</SupportedOSPlatformVersion>
     <TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</TargetPlatformMinVersion>
+
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <WindowsSdkPackageVersion>10.0.19041.41</WindowsSdkPackageVersion>
+    
+    <!-- Remove this NoWarn once .NET MAUI has implplemented partial classes for XamlUsreTypeInfo source generators -->
```
Comment (brminnick) [inline samples/CommunityToolkit.Maui.Markup.Sample/CommunityToolkit.Maui.Markup.Sample.csproj:34]:
```suggestion
    <!-- Remove this NoWarn once .NET MAUI has implplemented partial classes for XamlUserTypeInfo source generators -->
```

**Review by VladislavAntonyuk — DISMISSED** [review]

File: azure-pipelines.yml (hunk 2 above)
Comment (VladislavAntonyuk) [reply azure-pipelines.yml:70]: false

File: Directory.Build.props (hunk 5 above)
Comment (VladislavAntonyuk) [reply Directory.Build.props:7]: latest?

Comment (brminnick) [reply Directory.Build.props:7]: You're not wrong, but  I've updated all of the `[ObservableProperty]`s in the sample app to use partial properties. 

The MVVM toolkit uses the new `field` keyword for their implementations of partial properties, and in C# 13 we're still required to use `<LangVersion>preview</LangVersion>`  in order to access the new `field` keyword. 

Mads mentioned today that he plans on removing the `preview` requirement for the `field` keyword in C# 14.

**Comment by brminnick** [comment]
Thanks @VladislavAntonyuk! Good catch - I fixed the `allowsPrerelease` and am ready for a final review 👍

**Pull request merged**